### ISSUE TITLE
Initial support for lists

### ISSUE THREAD

**Issue by NuclearRedeye** [libero/editor#568]
# Description

Take a look at the available ProseMirror plugins for lists, and pick one to integrate and try.

# DoD
- [ ] A initial implementation of lists, where the user can create a list using the paragraph menu.

**Comment by fred-atherden** [comment]
Thanks! 

I think we also need lower/upper alpha/roman indicators? such as:
a.
A.
IX.
ix. 

We also need no indicator as well (so a simple list). These are outlined [here](https://github.com/libero/editor/issues/81#issuecomment-638119157). Maybe a question for @NickDuf as well. 

Also we need the ability to indent the list (lists within lists), and the user needs to be able to choose the indicators for this too (e.g. a list with lower alpha indicators in a list item for a numbered list).

**Comment by ilya-zhuravel** [comment]
@FAtherden-eLife Lists indentation works, but I couldn't set up **Tab** and **Shift-Tab**. Instead I used the default for prosemirror key bindings **Ctrl/Cmd + ]** and **Ctrl/Cmd + [**

Regarding lower upper alpha/roman indicator, current state is as per the epic. This needs further design and discussion as to how to toggle upper case and lower case.

**Comment by fred-atherden** [comment]
> @FAtherden-eLife Lists indentation works, but I couldn't set up Tab and Shift-Tab. Instead I used the default for prosemirror key bindings Ctrl/Cmd + ] and Ctrl/Cmd + [

Ah thanks. I recall you did mention that at a standup as well, sorry for forgetting. OK, yeah since this is just initial support, it looks good to me.

**Comment by NickDuf** [comment]
@FAtherden-eLife 
- Yes we did agreed on simple lists. If the user hits delete when they are at the start of a list item the indicator should disappear (like in g-docs or word)
- For mvp we decided not to be able to change list type indicators and go with what was on [this doc](https://docs.google.com/document/d/1FJAPEOOXsG5FLVf_eX7Sx4rI-m4tlBw29bhOncW67Jc/edit?pli=1). 

@ilya-zhuravel 
- Looks good to me. I think the ctrl/cmd will cause us a usability issue, but I'm happy to hold off and put it to the test if using tab will be difficult to implement
- Each item has a margin-bottom of 16px which is spacing them out too much. I would probably remove it and add the 16px to bottom of the ul or ol element.
- Switching to a new list type within a list should start at 1. or solid bullet and follow its indicator order
